utilities/transactions/pessimistic_transaction_db.h

```diff
@@ -172,7 +175,7 @@ class PessimisticTransactionDB : public TransactionDB {
   friend class WriteUnpreparedTransactionTest_RecoveryTest_Test;
   friend class WriteUnpreparedTransactionTest_MarkLogWithPrepSection_Test;
 
-  std::unique_ptr<LockManager> lock_manager_;
+  std::shared_ptr<LockManager> lock_manager_;
```

Comment:
who needs to share this lockmanager with?

Reply:
It shares it with TransactionDBOptions::lock_mgr_handle.
The workflow is:

* the user passes the manager in TransactionDBOptions::lock_mgr_handle
* TransactionDB gets another pointer to it in NewLockManager, here:

https://github.com/spetrunia/rocksdb/blob/1dfeb2bfa089adc78378459b40650b5b5f3288b7/utilities/transactions/lock/lock_manager.cc#L17